taskchampion/src/task/task.rs

```diff
@@ -263,9 +264,23 @@ impl<'r> TaskMut<'r> {
     /// Set the task's status.  This also adds the task to the working set if the
     /// new status puts it in that set.
     pub fn set_status(&mut self, status: Status) -> anyhow::Result<()> {
-        if status == Status::Pending {
-            let uuid = self.uuid;
-            self.replica.add_to_working_set(uuid)?;
+        if status == Status::Pending {}
+        match status {
+            Status::Pending => {
+                // clear "end" when a task becomes "pending"
+                if self.taskmap.contains_key(Prop::End.as_ref()) {
+                    self.set_timestamp(Prop::End.as_ref(), None)?;
+                }
+                let uuid = self.uuid;
+                self.replica.add_to_working_set(uuid)?;
+            }
+            Status::Completed | Status::Deleted => {
+                // set "end" when a task is deleted or completed
+                if !self.taskmap.contains_key(Prop::End.as_ref()) {
```

Comment:
Somewhat unrelated thing, but if `Prop` was `#[derive(Copy)]` would it neaten these checks to `if self.taskmap.contains_key(Prop::End)` ?

Reply:
The `as_ref` here is returning `&'static str`, rather than `&Prop` (it's a strum method).  So implementing Copy doesn't allow removing `.as_ref()`.

Reply:
Ahh, right!